llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -2041,6 +2041,28 @@ void CombinerHelper::applyCombineMulToShl(MachineInstr &MI,
   Observer.changedInstr(MI);
 }
 
+bool CombinerHelper::matchCombineSubToAdd(MachineInstr &MI,
+                                          BuildFnTy &MatchInfo) {
+  GSub &Sub = cast<GSub>(MI);
+
+  LLT Ty = MRI.getType(Sub.getReg(0));
+
+  if (!isLegalOrBeforeLegalizer({TargetOpcode::G_ADD, {Ty}}))
+    return false;
+
+  APInt Imm = getIConstantFromReg(Sub.getRHSReg(), MRI);
+
+  MatchInfo = [=, &MI](MachineIRBuilder &B) {
+    auto NegCst = B.buildConstant(Ty, -Imm);
```

Comment:
The check for G_CONSTANT is still missing.

Reply:
No please ignore. We found a G_CONSTANT by the type.

Reply:
I don't know what you mean. I checked G_CONSTANT in the .td file like you told me to.

Reply:
`B.buildConstant` builds a G_CONSTANT of type `Ty`. We must check that it is legal. But we found a G_CONSTANT on the rhs of the G_SUB. Thus G_CONSTANTs of  type `Ty` are legal or we are before the legalizer.